x-pack/legacy/plugins/infra/public/pages/metrics/components/section.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent, useCallback } from 'react';
+import { EuiTitle } from '@elastic/eui';
+import { SideNavContext, SubNavItem } from '../lib/side_nav_context';
+import { LayoutProps } from '../types';
+
+type SectionProps = LayoutProps & {
+  navLabel: string;
+  sectionLabel: string;
+};
+
+export const Section: FunctionComponent<SectionProps> = ({
+  children,
+  metrics,
+  navLabel,
+  sectionLabel,
+  onChangeRangeTime,
+  isLiveStreaming,
+  stopLiveStreaming,
+}) => {
+  const { addNavItem } = React.useContext(SideNavContext);
```

Comment:
Just out of curiosity, what's your philosophy on when to `import {useContext}` vs when to use the fully reference a static function `React.userContext` or `React.Children` etc?

Reply:
Normally I use the `import { useContext }` this happens only when I'm lazy or experimenting with an idea and forgot to fix it :D